x-pack/plugins/security_solution/public/management/components/endpoint_responder/components/offline_callout.tsx

```diff
@@ -5,27 +5,51 @@
  * 2.0.
  */
 
-import React, { memo } from 'react';
+import React, { memo, useMemo } from 'react';
 import { EuiCallOut, EuiSpacer } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n-react';
 import { i18n } from '@kbn/i18n';
+import { useIsExperimentalFeatureEnabled } from '../../../../common/hooks/use_experimental_features';
+import { useGetSentinelOneAgentStatus } from '../../../../detections/components/host_isolation/use_sentinelone_host_isolation';
+import type { ResponseActionAgentType } from '../../../../../common/endpoint/service/response_actions/constants';
 import { useGetEndpointDetails } from '../../../hooks';
 import { HostStatus } from '../../../../../common/endpoint/types';
 
 interface OfflineCalloutProps {
+  agentType: ResponseActionAgentType;
   endpointId: string;
+  hostName: string;
 }
 
-export const OfflineCallout = memo<OfflineCalloutProps>(({ endpointId }) => {
+export const OfflineCallout = memo<OfflineCalloutProps>(({ agentType, endpointId, hostName }) => {
+  const isEndpointAgent = agentType === 'endpoint';
+  const isSentinelOneAgent = agentType === 'sentinel_one';
+  const isSentinelOneV1Enabled = useIsExperimentalFeatureEnabled(
+    'responseActionsSentinelOneV1Enabled'
+  );
+
   const { data: endpointDetails } = useGetEndpointDetails(endpointId, {
     refetchInterval: 10000,
+    enabled: isEndpointAgent,
   });
 
-  if (!endpointDetails) {
+  const { data } = useGetSentinelOneAgentStatus([endpointId]);
+
+  const showOfflineCallout = useMemo(
+    () =>
+      (isEndpointAgent && endpointDetails?.host_status === HostStatus.OFFLINE) ||
+      (isSentinelOneAgent && data?.[endpointId].status === HostStatus.OFFLINE),
```

Comment:
FYI: When you start working on implementing support to the `agent status` api for all agent types (currently supports only sentinelone), remember to come back here and simplify this component to only look at one data source

Reply:
added a TODO